### ISSUE TITLE
RSC InL(?) crashing VCV Rack

### ISSUE THREAD

**Issue by firolightfog** [docb/dbRackModules#27]
Hello Christian,

I already mentioned this earlier (see https://community.vcvrack.com/t/dbrackmodules-releases/16147/10?u=firolfo ) but now (after a couple of months later) I still has the issue.

Plugging a cable into the *InL socket* I get a crash occasionally. I can't recreate the crash on demand but I believe that there's something wrong there in RSC.

Regards, A.

**Comment by docb** [comment]
Hi Andras,
it would be still good if i would have a log file when the crash occurs. i cannot reproduce this behavior on Linux.
how often is it happening?
best regards,
Christian

**Comment by firolightfog** [comment]
Well, pretty often. Like 50% or so. 

At the moment of plugging the cable into RSc the Rack is freezing and I see this in log.txt:

```
[7.735 info src/app/Browser.cpp:93 chooseModel] Creating module widget docB RSC
[7.737 info src/app/ModuleWidget.cpp:584 load] Loading preset C:/Users/aszabo/Documents/Rack2/presets/dbRackModules/RSC/template.vcvm
[7.737 info src/engine/Module.cpp:167 fromJson] Patch created with dbRackModules v2.0.5, currently using v2.1.0.
[7.825 info src/network.cpp:147 requestJson] Requesting JSON GET https://api.vcvrack.com/modules
[9.776 info src/window/Window.cpp:50 loadFile] Loaded font C:/Users/aszabo/OneDrive - UNICEF/private/DAW/Rack2Free/res/fonts/Nunito-Bold.ttf
[15.001 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[15.017 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
```

So I'm just forcing Rack window to close by pressing the X. However the noise indicates that Rack is still running.

I "End Task" in Task Manager and I see this at the end of the log.txt:

```
[15.001 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[15.017 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[30.002 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[30.015 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[45.018 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[45.028 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[60.019 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[60.027 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[75.019 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[75.027 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[90.019 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[90.032 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[105.020 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[105.027 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[120.028 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[120.038 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[135.030 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[135.037 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[150.037 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[150.048 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[165.037 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[165.045 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[180.038 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[180.046 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[183.524 info adapters/standalone.cpp:241 main] Stopped window
[183.524 info adapters/standalone.cpp:252 main] Deleting context
[183.524 info src/context.cpp:18 ~Context] Deleting window
[183.554 info src/context.cpp:22 ~Context] Deleting patch manager
[183.554 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[183.576 info src/context.cpp:26 ~Context] Deleting scene
```

Sorry, I know it's not much of a help...

**Comment by docb** [comment]
as far as i understood: you add RSC via the module browser and then you plug a cable into InL and then it freezes.

could you please remove the template and try again? this will most likely not solve but just in case .....

**Comment by docb** [comment]
you could then test this one:
https://github.com/docb/dbRackModules/releases/tag/v2.1.1-dev1
(also you can try out the eucledian rythm feature in HexSeq -right click on text field)

**Comment by firolightfog** [comment]
> as far as i understood: you add RSC via the module browser and then you plug a cable into InL and then it freezes.
> 
> could you please remove the template and try again? this will most likely not solve but just in case .....

I renamed my template.vcvm and got no freeze ever since. I'm not sure if it's coincident. Anyway this is the template I had before.

[template.vcvm.txt](https://github.com/docb/dbRackModules/files/9545929/template.vcvm.txt)
 
-----

I gave a quick try with the DEV1 release and without my template.vcvm but I got the same freeze again. Last lines of log.txt:

```
[41.852 info src/app/Browser.cpp:89 chooseModel] Creating module docB RSC
[41.854 info src/app/Browser.cpp:93 chooseModel] Creating module widget docB RSC
[45.038 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[45.046 info src/settings.cpp:470 save] Saving settings C:/Users/aszabo/Documents/Rack2/settings.json
[55.332 info adapters/standalone.cpp:241 main] Stopped window
[55.332 info adapters/standalone.cpp:252 main] Deleting context
[55.332 info src/context.cpp:18 ~Context] Deleting window
[55.371 info src/context.cpp:22 ~Context] Deleting patch manager
[55.371 info src/patch.cpp:211 saveAutosave] Saving autosave C:/Users/aszabo/Documents/Rack2/autosave/patch.json
[55.380 info src/context.cpp:26 ~Context] Deleting scene
```

Best regards, Andras

**Comment by firolightfog** [comment]
https://user-images.githubusercontent.com/34127628/189605612-a9ca8e78-4966-4577-8a48-d4705cf2c55e.mp4

**Comment by docb** [comment]
hmm this is interesting. the code in 2.1.1. did not change anything but initializing some variables with zero. so either the problem is not solved at all, or something is wired when upgrading to a new version.
could you confirm that 2.1.0 does still not reproduce the problem?

**Comment by docb** [comment]
https://github.com/docb/dbRackModules/releases/tag/v2.1.1-dev2
i found a state where i could reproduce the problem and i fixed it. Could you please give a try? thanks.

**Comment by firolightfog** [comment]
Quick results: I tried it in a few of my patches and dev2 looks good. 
Cheers, A.

https://user-images.githubusercontent.com/34127628/189687089-9cc7ec54-5ea4-48fb-98e4-ecdd308efac0.mp4

**Comment by docb** [comment]
great. what  modules are there in the first row (Sheep, ... etc) ?

**Comment by firolightfog** [comment]
> great. what modules are there in the first row (Sheep, ... etc) ?

Those are my unreleased prototypes. The ones in the video are [RV8](https://github.com/firolightfog/Yolow#RandVolt8), [PushMe](https://github.com/firolightfog/Yolow#PushMeSeq), [TRP1](https://github.com/firolightfog/Yolow#TrigPrec1), [Quant12](https://github.com/firolightfog/Yolow#Quant12) and Sheep. Sheep is a mutating 32 random map that can be read as a set of stepped voltages, or trigger sequence (by defining a breakpoint/threshold voltage). 

Whenever I feel that I'm too lazy to figure out how to do something with the existing modules, or my solution is way too big and complicated then I try to create a module myself. There are two obstacles only: (1) I'm not good at coding, (2) usually I don't know what I want. You can see the signs of both if you ever dare to take a look ... :)

**Comment by docb** [comment]
fixed in 2.1.3
